tests/validate_datasets.py

```diff
@@ -0,0 +1,45 @@
+import intake
+import pytest
+
+def crawl_nested_catalog(cat_or_entry, parent_name=()):
```

Comment:
May be simpler to use [`cat.walk`](https://intake.readthedocs.io/en/latest/api_user.html#intake.catalog.local.YAMLFileCatalog.walk)